tests/bridge_message_tests.cpp

```diff
@@ -325,4 +325,88 @@ BOOST_FIXTURE_TEST_CASE(test_send_message_from_solidity, bridge_message_tester)
   }
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(handler_tests, bridge_message_tester) try {
+  auto emv_reserved_address = make_reserved_address(evm_account_name);
+
+  // Fund evm1 address with 1001 EOS
+  evm_eoa evm1;
+  const int64_t to_bridge = 1001'0000;
+  transfer_token("alice"_n, evm_account_name, make_asset(to_bridge), evm1.address_0x());
+
+  BOOST_REQUIRE_EXCEPTION(send_raw_message(evm1, emv_reserved_address, 0,
+                                                     evmc::from_hex(bridgeMsgV0_method_id).value() +
+                                                     evmc::from_hex(int_str32(96)).value() +
+                                                     evmc::from_hex(int_str32(1)).value() +
+                                                     evmc::from_hex(int_str32(160)).value() +
+                                                     evmc::from_hex(int_str32(8)).value() +
+                                                     evmc::from_hex(data_str32(str_to_hex("receiver"))).value() +
```

Comment:
Why not refactor `send_bridge_message` to generate the call data separately from sending the transaction? That way you can call the function to generate the call data here and elsewhere rather than duplicating this fragile code.

Reply:
fixed